lib/bolt/logger.rb

```diff
@@ -123,18 +131,59 @@ def self.reset_logging
       Logging.reset
     end
 
-    def self.warn_once(type, msg)
+    # Issues a warning unless the user has suppressed it.
+    #
+    def self.warn(id:, msg:, src: self)
```

Comment:
If you made this method take a block which is expected to return the message when yielded (the other log methods support that syntax as well), then you could use `block.binding.receiver` to get the caller. Kind of weird, but should work.

Reply:
Hrm... that'd require calling the method with a block, wouldn't it? I.e.

```
Bolt::Logger.warn(...) do; end
```

Reply:
Yeah it would be something like `Bolt::Logger.warn('problem_occurred') { "Something terrible has happened" }`